t/base/protocols/file.t

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/perl
+use utf8;
+use strict;
+use warnings;
+use Test::More tests => 4;
+
+use File::Basename qw(dirname);
+use LWP::Simple qw(get);
+use File::Temp qw(tempfile);
+
+require POSIX;
+POSIX::setlocale(&POSIX::LC_ALL, 'ja_JP.UTF-8');
+
+my $tmp, my $tmpd;
+$tmpd = File::Temp->newdir();
+$tmpd->unlink_on_destroy(1);
+$tmp = File::Temp->new(TEMPLATE => 'wwwdata_tempXXXX',
+                       DIR => $tmpd,
+                       SUFFIX => '.コピペ.test');
+
+open(TMP, '>:utf8', $tmp);
+print TMP 'テスト';
+close(TMP);
+
+## Test default directory output.
+is($tmp->filename =~ /コピペ/, 1);
+my $res = get('file://' . dirname($tmp->filename));
+is(($res =~ m/コピペ/, $&), 'コピペ');
+# Make sure there are no HTML entities.
+isnt(($res =~ m/&/, $&), '&');
```

Comment:
Same as above. Did you mean "this should not contain an ampersand"? I'd write that as `unlike($res, qr/&/, 'output has no HTML entities')`. I suspect the bit you don't know about here is the `qr//`, which constructs a regular expression without applying it. That way you can pass a pattern around, and the `like` and `unlike` will apply it later. Here's a link to the doc: https://metacpan.org/pod/Test::More#like